node/src/components/linear_chain.rs

```diff
@@ -136,12 +136,12 @@ impl<I> LinearChain<I> {
     }
 
     /// Adds pending finality signatures to the block; returns events to announce and broadcast
-    /// them.
+    /// them, and the updated block.
     fn add_pending_finality_signatures<REv>(
         &mut self,
-        block: &mut Block,
+        mut block: Block,
         effect_builder: EffectBuilder<REv>,
-    ) -> Effects<Event<I>>
+    ) -> (Box<Block>, Effects<Event<I>>)
```

Comment:
Why does it need to be `Box`-ed?

Reply:
Because it gets put into events where it's boxed.
Clippy often complains about enums with variants of very different sizes because the small variants use as much memory as the large ones.

Reply:
Ah, right. I would have boxed it on the caller side though, since that's where we know it has to be boxed. If it wasn't put into an event clippy would not complain.

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/649/commits/be5e687d5c2c5b4003edf4c09bd5becea9df1402.